### PR TITLE
Move tagging of images to registry.go

### PR DESCRIPTION
**Pull request by a-cool-train**
*Description of changes:*
- Currently Replacement of default registry and curated packages prod/dev registries occur in docker.go. Ideally we should move this replacement to the registry.go.

*Testing (if applicable):*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pkg/docker/registry.go [hunk 1]
```diff
@@ -104,6 +105,13 @@ func getUpdatedEndpoint(originalEndpoint, image string) string {
 	return originalEndpoint
 }
 
+// Replace original registries with the new endpoint from the registry mirror configuration.
+func getLocalImage(endpoint string, image string) string {
+	replacer := strings.NewReplacer(defaultRegistry, endpoint, packageProdDomain, endpoint, packageDevDomain, endpoint)
```
Comment (g-gaston) [inline pkg/docker/registry.go:110]: curious, why not just always replace the host, irregardless of which one it is? That way we make this agnostic of the different ecr domains, it just knows that it need to replace the domain with `endpoint`

Comment (a-cool-train) [reply pkg/docker/registry.go:110]: Not sure if there was an implementation detail here. I just added extra endpoints to an already existing functionality. I can try urls.replacehost and see if that would work the same mechanism.

Comment (a-cool-train) [reply pkg/docker/registry.go:113]: Looks like there is an issue with url parsing that causes an issue. I put a comment on the code to indicate the issue.

**Comment by jonahjon** [comment]
Looks like a failing test, they might need to be tweaked.

**Comment by TerryHowe** [comment]
The copy packages command should make this unnecessary. There is some clean-up here that should be done, but that looks a lot different.

**Pull request closed without merging**